app/controllers/api/ladder_matches_controller.rb

```diff
@@ -0,0 +1,57 @@
+class Api::LadderMatchesController < ApplicationController
+  def index
+    matches = LadderMatch.all
+    render json: {
+      matches: JSON.parse(matches.to_json())
+    }
+  end
+  def create
+    p = ActionController::Parameters.new params[:ladder_match]
+    permitted = p.permit(:player1,:player2, :date_of_match, :winner)
+    match = LadderMatch.new permitted
+    @player1 = LadderUser.find_by(username: p[:player1])
+    match.password = p[:password]
+	#authentication and validations on serverside
+    unless match.valid? && @player1.try(:authenticate, match.password)
+      unless @player1.try(:authenticate, match.password)
+        match.errors.add(:password, 'This player (1) and password combination is invalid')
+      end
+      render json: { errors: match.errors }, status: 400 and return
+    end
+    #Point Manipulation & updates.
+    @player2 = LadderUser.find_by(username: p[:player2])
+    match.points = update_players(@player1,@player2,match.winner,match.id).abs
+    match.save!
+    #verification mail.
+    LadderMailer.match_mail(match,@player1,@player2).deliver!
+    render json: {ladder_match: match}
+  end  
+
+  def update_players(player1,player2,winner,id)
+    #Match level calculations, using match's class level methods
+    elo_notWLD = true
```

Comment:
It's for future usage - If you want to make a ladder that's WLD compared to Elo style, you just switch the constant to false.